lib/services/elasticsearch.js

```diff
@@ -153,1195 +168,1048 @@ class ElasticSearch extends Service {
    *
-   * @param {Request} request
-   * @return {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   *
+   * @returns {Promise.<Object>} { }
    */
-  updateMapping(request) {
-    const esRequest = initESRequest(request);
+  truncateCollection (index, collection) {
+    let mappings;
 
-    return this.esWrapper.getMapping(esRequest, true)
-      .then(mappings => {
-        const collectionMapping =
-          mappings[esRequest.index].mappings[esRequest.type];
-        const requestBody = request.input.body || {};
-
-        this._checkMapping(requestBody);
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
 
-        esRequest.body = this._mergeDefaultMapping(
-          esRequest.index,
-          collectionMapping,
-          requestBody
-        );
+    return this.getMapping(index, collection)
+      .then(collectionMappings => {
+        mappings = collectionMappings;
 
-        return this.client.indices.putMapping(esRequest)
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      });
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => this.client.indices.create({ ...esRequest, mappings }))
+      .then(() => {
+        // return nothing
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Retrieve mapping definition for index/type
+   * Runs several action and document
    *
-   * @param {Request} request
-   * @return {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Array<Object>} documents - Documents to import
+   * @param {Object} options - timeout (null), refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { items, errors }
    */
-  getMapping(request, includeKuzzleMeta = false) {
-    const esRequest = initESRequest(request);
+  import (
+    index,
+    collection,
+    documents,
+    { refresh=false, timeout=null, userId=null } = {})
+  {
+    const
+      esIndex = this._getESIndex(index, collection),
+      actionNames = ['index', 'create', 'update', 'delete'],
+      dateNow = Date.now(),
+      esRequest = {
+        body: documents,
+        refresh,
+        timeout
+      },
+      kuzzleMeta = {
+        created: {
+          author: getUserId(userId),
+          createdAt: dateNow,
+          updatedAt: null,
+          updater: null
+        },
+        updated: {
+          updater: getUserId(userId),
+          updatedAt: dateNow
+        }
+      };
 
-    return this.esWrapper.getMapping(esRequest, includeKuzzleMeta);
-  }
+    assertWellFormedRefresh(esRequest);
 
-  /**
-   * Retrieve the complete list of existing data collections in the current index
-   *
-   * @param {Request} request
-   * @return {Promise}
-   */
-  listCollections(request) {
-    const esRequest = initESRequest(request);
+    let lastAction;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "lastAction" to line 768. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2392)